### ISSUE TITLE
ansible-vault

### ISSUE THREAD

**Issue by netw0rm** [ansible/ansible#19432]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ansible-vault

##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file =
  configured module search path = Default w/o overrides
```

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```
after ansible-vault create hosts,  ansible -i hosts all -m ping  doesn't work
```

**Comment by bcoca** [comment]
can you explain in more detail or give a reproducible example?

**Comment by netw0rm** [comment]
I typed  ansible -I hosts --ask-vault-pass all -m ping ,it returned this:

$ANSIBLE_VAULT;1.1;AES256 | UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh: ssh: Could not resolve hostname $ansible_vault;1.1;aes256: Name or service not known\r\n",
    "unreachable": true
}
...

**Comment by bcoca** [comment]
not nearly enough information, please complete the template above and provide steps/setup you have to get to that point

**Comment by global** [comment]
Hi, I think @netw0rm  is trying to encrypt the inventory file using ansible-vault, which is not possible AFAIK. 

@netw0rm check on the page (http://docs.ansible.com/ansible/playbooks_vault.html#what-can-be-encrypted-with-vault) what can be encrypted.

**Comment by bcoca** [comment]
@global encrypted inventory is supported

**Comment by mdvickst** [comment]
I have been encrypting my inventory files and it was worked fine up until 2.2 was released. I may be reading the docs incorrectly but it says that you can encrypt "any structured data file used by Ansible". Isn't the inventory file a structured data file?

I just verified that it works to encrypt/decrypt the inventory file with ansible 2.0 on Fedora 24 and it was working with 2.1 as well. 

My use case is that I have a program that uses ansible core under the covers to perform discovery and it builds an inventory file with ansible_ssh_user and ansible_ssh_pass defined in the inventory file under a [all:vars] section. I realize I could have split those variables out into a separate file but I would need to re-write my script that has already been distributed. 

Could someone please explain the reason why encrypting the inventory is not supported and why that previous functionality should be removed?

**Comment by bcoca** [comment]
dupe of already fixed #18355
